tests/golos.publication_rewards_tests.cpp

```diff
@@ -880,4 +951,39 @@ BOOST_FIXTURE_TEST_CASE(golos_curation_test, reward_calcs_tester) try {
     check();
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(golos_close_acc_test, reward_calcs_tester) try {
+    BOOST_TEST_MESSAGE("golos_close_acc_test");
+    int64_t maxfp = std::numeric_limits<fixp_t>::max();
+    auto bignum = 500000000000;
+    init(bignum, 500000);
+    produce_blocks();
+    BOOST_TEST_MESSAGE("--- setrules");
+    using namespace golos_curation;
+    BOOST_CHECK_EQUAL(success(), setrules({"x", maxfp}, {golos_curation::func_str, maxfp}, {"1", bignum},
+        [](double x){ return x; }, golos_curation::func, [](double x){ return 1.0; }));
+    check();
+
+    BOOST_TEST_MESSAGE("--- add_funds_to_forum");
+    BOOST_CHECK_EQUAL(success(), add_funds_to_forum(50000));
+    check();
+
+    BOOST_TEST_MESSAGE("--- checking payout if account is closed");
+    BOOST_TEST_MESSAGE("--- create_message: " << name{_users[0]}.to_string());
+    BOOST_CHECK_EQUAL(success(), create_message({_users[0], "permlink"}));
+    check();
+
+    BOOST_TEST_MESSAGE("--- " << name{_users[1]}.to_string() << " voted");
+    BOOST_CHECK_EQUAL(success(), addvote(_users[1], {_users[0], "permlink"}, 10000));
+    check();
+
+    BOOST_CHECK_EQUAL(success(), retire_vest(vest.get_balance_raw(_users[0])["vesting"].as<asset>(), _users[0], _issuer));
+    check();
+
+    BOOST_CHECK_EQUAL(success(), close_token_acc(_users[0], _token_symbol));
+    BOOST_CHECK_EQUAL(success(), close_vest_acc(_users[0], _token_symbol));
+
+    produce_blocks(golos::seconds_to_blocks(post.window));
+    check();
```

Comment:
The second problem: if test fails, we don't know, is vesting or token part broken. So it's better to have separate tests: 1) only token balance closed; 2) only vesting balance closed